packages/mui-material/src/SvgIcon/SvgIcon.js

```diff
@@ -85,28 +85,72 @@ const SvgIcon = React.forwardRef(function SvgIcon(inProps, ref) {
   };
 
   const more = {};
+  let newComponent;
+  let hasTitleAccess = titleAccess !== undefined;
 
   if (!inheritViewBox) {
     more.viewBox = viewBox;
   }
 
   const classes = useUtilityClasses(ownerState);
 
+  if (typeof component === 'function') {
+    const element = component({});
```

Comment:
I don't understand why we need this.

Reply:
It happens When we use component prop as a function Component. We need to merge the children (https://github.com/mui/material-ui/blob/master/packages/mui-material/src/SvgIcon/SvgIcon.js#L108-L109) into the function Component.